R/mod_body_coc_competition.R

```diff
@@ -0,0 +1,475 @@
+        
+        psh <- estimated_score |> dplyr::left_join(dq, by = "Measure") |>
+          dplyr::ungroup() |>
+          dplyr::left_join(possible_score, by = "Measure") |>
+          dplyr::left_join(calculation, by = "Measure") |>
+          dplyr::mutate(
+            DQ = dplyr::case_when(
+              DQflag == 0 ~ "Data Quality passes",
+              DQflag == 1 ~ "Please correct your Data Quality issues so this item
+            can be scored",
+            DQflag == 2 ~ "", # "Documents not yet received",
+            DQflag == 3 ~ "", # "Docs received, not yet scored",
+            DQflag == 4 ~ "", # "CoC Error",
+            DQflag == 5 ~ "" # "Docs received past the due date"
+            )
+          ) |>
+          dplyr::filter(!Measure %in% c("Moved into Own Housing",
+                                 "Average Length of Stay")) |>
+          dplyr::select(1, Calculation, 2, "Possible Score" = 4, "Data Quality" = DQ)
+        
+        rrh <- estimated_score |> dplyr::left_join(dq, by = "Measure") |>
+          dplyr::ungroup() |>
+          dplyr::left_join(possible_score, by = "Measure") |>
+          dplyr::left_join(calculation, by = "Measure") |>
+          dplyr::mutate(
+            DQ = dplyr::case_when(
+              DQflag == 0 ~ "Data Quality passes",
+              DQflag == 1 ~ "Please correct your Data Quality issues so this item
+            can be scored",
+            DQflag == 2 ~ "", # "Documents not yet received",
+            DQflag == 3 ~ "", # "Docs received, not yet scored",
+            DQflag == 4 ~ "", # "CoC Error",
+            DQflag == 5 ~ "" # "Docs received past the due date"
+            )
+          ) |>
+          dplyr::filter(!Measure %in%
+                   c("Long Term Homeless",
+                     "Prioritization of Chronic",
+                     "Prioritization Workgroup")) |>
+          dplyr::select(1, Calculation, 2, "Possible Score" = 4, "Data Quality" = DQ)
+        
+        th <- estimated_score |> dplyr::left_join(dq, by = "Measure") |>
+          dplyr::ungroup() |>
+          dplyr::left_join(possible_score, by = "Measure") |>
+          dplyr::left_join(calculation, by = "Measure") |>
+          dplyr::mutate(
+            DQ = dplyr::case_when(
+              DQflag == 1 ~ "Please correct your Data Quality issues so this item
+            can be scored",
+            DQflag == 0 ~ "Data Quality passes",
+            DQflag == 2 ~ "", # "Documents not yet received",
+            DQflag == 3 ~ "", # "Docs received, not yet scored",
+            DQflag == 4 ~ "", # "CoC Error",
+            DQflag == 5 ~ "" # "Docs received past the due date"
+            )
+          ) |>
+          dplyr::filter(!Measure %in% c(
+            "Long Term Homeless",
+            "Prioritization of Chronic",
+            "Prioritization Workgroup"
+          )) |>
+          dplyr::select(1, Calculation, 2, "Possible Score" = 4, "Data Quality" = DQ)
+
+        DT::datatable(
```

Comment:
See `?datatable_default` in place of all calls to `DT::datatable`.
It's best practice to wrap any repeated actions in a wrapper function and only use the wrapped function such that if you need to change the way the wrapped function is expressed, or if the package maintainers change it, you can update it in a single place.

`datatable_default` also adds a bunch of helpful default optional features to the datatables for users.